fixit/rules/rewrite_to_literal.py

```diff
@@ -115,27 +146,44 @@ def visit_Call(self, node: cst.Call) -> None:
                     )
                     return
                 new_node = cst.Set(elements=elements)
-            elif len(elements) == 0 or m.matches(
-                exp.args[0].value,
-                m.Tuple(elements=[pairs_matcher]) | m.List(elements=[pairs_matcher]),
+            elif (
+                len(elements) == 0
+                or m.matches(
+                    exp.args[0].value,
+                    m.Tuple(elements=[pairs_matcher])
+                    | m.List(elements=[pairs_matcher]),
+                )
+                or keyword_dict
             ):
-                new_node = cst.Dict(
-                    elements=[
-                        (
-                            lambda val: cst.DictElement(
-                                val.elements[0].value, val.elements[1].value
+                if keyword_dict:
+                    new_node = cst.Dict(
+                        elements=[
+                            cst.DictElement(
+                                cst.SimpleString(f'"{arg.keyword.value}"'),
+                                arg.value,
                             )
-                        )(
-                            cst.ensure_type(
-                                ele.value,
-                                cst.Tuple
-                                if isinstance(ele.value, cst.Tuple)
-                                else cst.List,
+                            for arg in exp.args
+                            if arg.keyword
+                        ]
+                    )
+                else:
+                    new_node = cst.Dict(
+                        elements=[
+                            (
+                                lambda val: cst.DictElement(
+                                    val.elements[0].value, val.elements[1].value
+                                )
+                            )(
+                                cst.ensure_type(
+                                    ele.value,
+                                    cst.Tuple
+                                    if isinstance(ele.value, cst.Tuple)
+                                    else cst.List,
+                                )
```

Comment:
Though this is existing code but it looks weird that `ensure_type` is used with `isinstance`.
`ensure_type` is for verifying the type and helps the type checker understanding the type. https://libcst.readthedocs.io/en/latest/helpers.html?highlight=ensure_type#libcst.helpers.ensure_type
The large `elif` branch can be separated as multiple smaller elif to handle different cases separately. That way, we can use different matcher of Tuple and List metches in different elif and remove the isinstance. It can also simplify the `if keyword_dict` to check it only once.
```
elif (
                len(elements) == 0
                or m.matches(
                    exp.args[0].value,
                    m.Tuple(elements=[pairs_matcher])
                    | m.List(elements=[pairs_matcher]),
                )
                or keyword_dict
            ):
```